### PR TITLE
Use classnames

### LINKED ISSUE #2686: Use classnames library
In many places in the code we use `["class1", "class2"].join(" ")` pattern to create `className` prop value. It leads to hard-to-read code and leaves blank space at the end of the string if last class depends on some kind of check.

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it fixes ugly code in creating `className` prop values.
Fixes #2686 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).

### CODE REVIEW

File: saleor/static/dashboard-next/AppRoot.tsx [hunk 1]
```diff
@@ -376,10 +377,9 @@ export const AppRoot = decorate(
                                       {user.email}
                                     </Typography>
                                     <ArrowDropdown
-                                      className={[
-                                        classes.arrow,
-                                        menuOpen ? classes.rotate : undefined
-                                      ].join(" ")}
+                                      className={classNames(classes.arrow, {
```
Comment (patrys) [inline saleor/static/dashboard-next/AppRoot.tsx]: What about the alternative:

```typescript
classNames({
  [classes.arrow]: true,
  [classes.rotate]: menuOpen
})
```

File: saleor/static/dashboard-next/collections/components/CollectionDetails/CollectionDetails.tsx [hunk 2]
```diff
@@ -50,7 +51,9 @@ const CollectionDetails = decorate<CollectionDetailsProps>(
             <img src={collection.backgroundImage} className={classes.image} />
             <Typography
               variant="caption"
-              className={!!onImageRemove && !disabled ? classes.link : ""}
+              className={classNames({
+                [classes.link]: !!onImageRemove && !disabled
```
Comment (patrys) [inline saleor/static/dashboard-next/collections/components/CollectionDetails/CollectionDetails.tsx]: I think the leading `!!` is a no-op?

File: saleor/static/dashboard-next/collections/components/CollectionList/CollectionList.tsx [hunk 3]
```diff
@@ -81,7 +82,9 @@ const CollectionList = decorate<CollectionListProps>(
                       ? onRowClick(collection.id)
                       : undefined
                   }
-                  className={collection && onRowClick && classes.link}
+                  className={classNames({
+                    [classes.link]: !!collection
```
Comment (patrys) [inline saleor/static/dashboard-next/collections/components/CollectionList/CollectionList.tsx]: Is `!!` needed?

File: saleor/static/dashboard-next/AppRoot.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/AppRoot.tsx]: Yeah, I think it would improve readability.

**Pull request merged**